cmake/cmaize/project/impl_/add_target.cmake

```diff
@@ -0,0 +1,99 @@
+# Copyright 2023 CMakePP
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+include_guard()
+
+#[[[
+# Add a target to the project. Duplicate objects will not be added.
+#
+# This internal implementation exists so a required keyword argument is
+# not part of the public interface, as well as to handle both ``desc`` and
+# ``target`` types. Both types are effectively strings representing target
+# names in this algorithm and can be treated equivalently, but cannot be
+# typecast to each other by CMakePPLang. The CMakePPLang type checking
+# is bypassed through the aforementioned required keyword argument for
+# the target name, essentially combining the two types.
+#
+# :param __at_target: CMaizeTarget object to be added.
+# :type __at_target: CMaizeTarget
+# :param NAME: Required keyword argument. See description below.
```

Comment:
As a quick note here, the NAME kwarg workaround may not actually be necessary anymore. I think that it was fixed in CMakePPLang so that giving the `str` type would allow for both `desc` and `target` types on an argument.